### PR TITLE
show transport time in PHIR

### PR DESCRIPTION
**Pull request by Asa-Kosto-QTM**
Show transport time in PHIR with configurable parameters found in qtm_machine.py

### CODE REVIEW

File: pytket/phir/phirgen_parallel.py [hunk 1]
```diff
@@ -280,6 +280,31 @@ def format_and_add_primary_commands(
...
+    adjustment = 0.0
+    for op in ops:
+        if "qop" in op:
+            match op["qop"]:
+                case "RZ" | "R1XY":
+                    adjustment += machine.sq_time
+                case "RZZ":
+                    adjustment += machine.tq_time
+                case "Measure":
+                    adjustment += machine.meas_prep_time
```
Comment (qartik) [inline pytket/phir/phirgen_parallel.py:294]: Can there be any other case here? If so, we should handle that/include a warning.

File: pytket/phir/phirgen_parallel.py [hunk 2]
```diff
@@ -280,6 +280,31 @@ def format_and_add_primary_commands(
...
+        if "block" in op and op["block"] == "qparallel":
+            first_op = op["ops"][0]["qop"]
+            match first_op:
+                case "RZ" | "R1XY":
+                    adjustment += machine.sq_time
+                case "RZZ":
+                    adjustment += machine.tq_time
```
Comment (qartik) [inline pytket/phir/phirgen_parallel.py:301]: Can there be any other case here? If so, we should handle that/include a warning.

File: pytket/phir/phirgen_parallel.py [hunk 3]
```diff
@@ -289,8 +314,8 @@ def genphir_parallel(
         inp: list of shards
         machine: a QTM machine on which to simulate the circuit
     """
-    max_parallel_tq_gates = len(machine.tq_options)
-    max_parallel_sq_gates = len(machine.sq_options)
+    max_parallel_tq_gates = int(len(machine.tq_options) / 2)
+    max_parallel_sq_gates = int(len(machine.sq_options) / 2)
```
Comment (qartik) [inline pytket/phir/phirgen_parallel.py:318]: Would the following variant using floor division be the same?
```suggestion
    max_parallel_tq_gates = len(machine.tq_options) // 2
    max_parallel_sq_gates = len(machine.sq_options) // 2
```

File: pytket/phir/qtm_machine.py [hunk 4]
```diff
@@ -27,16 +27,11 @@ class QtmMachine(Enum):
         size=20,
         gateset=QTM_DEFAULT_GATESET,
         tq_options={0, 2, 4, 6, 8, 10, 12, 14, 16, 18},
-        # need to get better timing values for below
-        # but will have to look them up in hqcompiler
-        timings=MachineTimings(tq_time=3.0, sq_time=1.0, qb_swap_time=2.0),
-    ),
-    QtmMachine.H1_2: Machine(
```
Comment (qartik) [inline pytket/phir/qtm_machine.py:34]: I missed the rationale behind deleting this machine, do the same values apply for `H1_1` and `H1_2`? If so, does it make sense to rename the `H1_1` machine to just `H1`?

**Review by qartik — CHANGES_REQUESTED** [review]

File: pytket/phir/qtm_machine.py (hunk 4 above)
Comment (Asa-Kosto-QTM) [reply pytket/phir/qtm_machine.py:34]: the H1_1 and H1_2 machine objects were going to be identical within pytket-phir, so there was no reason to include 2 classes that would have exactly the same parameters

Comment (Asa-Kosto-QTM) [reply pytket/phir/qtm_machine.py:34]: it does make sense to just call it 'H1' in that case

**Review by qciaran — CHANGES_REQUESTED** [review]
LGTM. Results on my end are as expected. 

Just curious about the CLI handling if given a machine not currently supported.

File: pytket/phir/cli.py [hunk 5]
```diff
@@ -66,10 +66,8 @@ def main() -> None:
             circuit = circuit_from_qasm(file)
 
         match args.machine:
-            case "H1-1":
-                machine = QtmMachine.H1_1
-            case "H1-2":
-                machine = QtmMachine.H1_2
+            case "H1-1" | "H1-2" | "H1":
+                machine = QtmMachine.H1
```
[the remark this answers is not in the document]
Comment (qartik) [reply pytket/phir/cli.py:70]: I shared a patch with @Asa-Kosto-QTM earlier. The only choices allowed here are those defined at https://github.com/CQCL/pytket-phir/pull/147/files/c10fa500181191dfca6300cb41b9d30d5ecd9b75#diff-78dd65ee7226caae6efcb9cf01811fba4e6caa2a8d795bc81fac94731ce59336R46, ie, "H1" won't ever appear and neither will anything else unless the choices in the CLI parser are updated.

Here's an error, e.g., after applying my patch:
```
❯ phirc -m H2 tests/data/qasm/baby.qasm
usage: phirc [-h] [-w WASM_FILE] [-m {H1}] [-v] [--version] qasm_files [qasm_files ...]
phirc: error: argument -m/--machine: invalid choice: 'H2' (choose from 'H1')
```

**Review by qartik — APPROVED** [review]

**Review by qciaran — APPROVED** [review]

**Pull request merged**